kafka-plugins-0.10/pom.xml

```diff
@@ -48,6 +52,10 @@
           <groupId>org.apache.spark</groupId>
           <artifactId>spark-tags_2.11</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>net.jpountz.lz4</groupId>
```

Comment:
why is this exclusion needed?

Reply:
without exclusion we hit this exception:

`java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: net.jpountz.lz4.LZ4BlockInputStream.<init>(Ljava/io/InputStream;Z)V
        at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:306) ~[hive-exec-1.2.1.jar:1.2.1]
        at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:293) ~[hive-exec-1.2.1.jar:1.2.1]
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116) ~[hive-exec-1.2.1.jar:1.2.1]
        at io.cdap.cdap.app.runtime.spark.SparkRuntimeService.run(SparkRuntimeService.java:346) ~[cdap-spark-core2_2.11-6.4.0-SNAPSHOT.jar:na]
        at com.google.common.util.concurrent.AbstractExecutionThreadService$1$1.run(AbstractExecutionThreadService.java:52) ~[hive-exec-1.2.1.jar:1.2.1]
        at io.cdap.cdap.app.runtime.spark.SparkRuntimeService$5$1.run(SparkRuntimeService.java:404) [cdap-spark-core2_2.11-6.4.0-SNAPSHOT.jar:na]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
Caused by: java.lang.NoSuchMethodError: net.jpountz.lz4.LZ4BlockInputStream.<init>(Ljava/io/InputStream;Z)V
        at org.apache.spark.io.LZ4CompressionCodec.compressedInputStream(CompressionCodec.scala:122) ~[spark-core_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$$anonfun$deserialize$2.apply(Checkpoint.scala:169) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$$anonfun$deserialize$2.apply(Checkpoint.scala:162) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377) ~[spark-core_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.Checkpoint$.deserialize(Checkpoint.scala:175) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]
        at org.apache.spark.streaming.CheckpointReader$$anonfun$read$2.apply(Checkpoint.scala:356) ~[spark-streaming_2.11-2.3.0.jar:2.3.0]`

Reply:
I see. That seems like a class conflict. Does it only happen for 2.3 spark version and not for 2.2? 
Also could you please share results for `mvn dependency:tree` without this exclusion via email?